src/include/sof/math/trig.h

```diff
@@ -137,17 +156,139 @@ static inline int16_t sin_fixed_16b(int32_t th_rad_fxp)
+ * | thRadFxp	      |cdccexpth        |thRadFxp|   cdccexpth|
+ * +----+-----+-------+----+----+-------+--------+------------+
+ * |WLen| FLen|Signbit|WLen|FLen|Signbit| Qformat| Qformat    |
+ * +----+-----+-------+----+----+-------+--------+------------+
+ * | 32 | 28  |  1    | 32 | 15 |   1	| 4.28	 | 2.30       |
+ * +------------------+-----------------+--------+------------+
+ */
+static inline struct cmpx cmpx_exp_32b(int32_t th_rad_fxp)
+{
+	int32_t th_cdc_fxp;
+	int32_t sign;
+	int32_t b_yn;
+	int32_t xn;
+	struct cmpx cexp;
+
+	cordic_approx(th_rad_fxp, CORDIC_31B_TABLE_SIZE, &sign, &b_yn, &xn, &th_cdc_fxp);
+	cexp = cmpx_cexp(sign, b_yn, xn, EN_32B_CORDIC_CEXP);
+
+	return cexp;
+}
+
+/**
+ * CORDIC-based approximation of complex exponential e^(j*THETA).
+ * computes COS(THETA) + j*SIN(THETA) using CORDIC algorithm
+ * approximation and returns the complex result.
+ * THETA values must be in the range [-2*pi, 2*pi). The cordic
+ * exponential algorithm converges, when the angle is in the
+ * range [-pi/2, pi/2).If an angle is outside of this range,
+ * then a multiple of pi/2 is added or subtracted from the
+ * angle until it is within the range [-pi/2,pi/2).Start
+ * with the angle in the range [-2*pi, 2*pi) and output has
+ * range in [-1.0 to 1.0]
+ * +------------------+-----------------+--------+------------+
+ * | thRadFxp	      |cdccexpth        |thRadFxp|   cdccexpth|
+ * +----+-----+-------+----+----+-------+--------+------------+
+ * |WLen| FLen|Signbit|WLen|FLen|Signbit| Qformat| Qformat    |
+ * +----+-----+-------+----+----+-------+--------+------------+
+ * | 32 | 28  |  1    | 32 | 15 |   1	| 4.28	 | 17.15       |
+ * +------------------+-----------------+--------+------------+
+ */
+static inline struct cmpx cmpx_exp_16b(int32_t th_rad_fxp)
+{
+	int32_t th_cdc_fxp;
+	int32_t sign;
+	int32_t b_yn;
+	int32_t xn;
+	struct cmpx cexp;
+
+	/* compute coeff from angles */
+	cordic_approx(th_rad_fxp, CORDIC_15B_TABLE_SIZE, &sign, &b_yn, &xn, &th_cdc_fxp);
+	cexp = cmpx_cexp(sign, b_yn, xn, EN_16B_CORDIC_CEXP);
+	/* return the complex result */
+	return cexp;
+}
+
+/**
+ * CORDIC-based approximation of inverse cosine
+ * inverse cosine of cdc_acos_th based on a CORDIC approximation
+ * acos(X) inverse cosine angle values in radian produces using the DCORDIC
+ * (Double CORDIC) algorithm.
+ * Q2.30 cdc_acos_th , input value range [-1 to 1]
+ * Q2.30 th_rad_fxp, output value range [3.14159265346825 to 0]
+ * LUT size set type 31
+ * Error (max = 0.000000026077032), THD+N  = -157.948952635422842
+ */
+static inline int32_t acosine_fixed_32b(int32_t cdc_acos_th)
+{
+	int32_t saturated_unary_minus;
+	int32_t th_rad_fxp;
+
+	saturated_unary_minus = cdc_acos_th;
+	if (saturated_unary_minus >= 0) {
+		th_rad_fxp = is_scalar_cordic(saturated_unary_minus, CORDIC_31B_TABLE_SIZE,
+					      cordic_ilookup, EN_32B_CORDIC_ACOSINE);
+	} else {
+		saturated_unary_minus = -saturated_unary_minus;
```

Comment:
the invert can be done for argument, `is_scalar_cordic(-saturated_unary_minus, ..`.